test/functional/admin/edition_lead_images_controller_test.rb

```diff
@@ -0,0 +1,53 @@
+require "test_helper"
+
+class Admin::EditionLeadImagesControllerTest < ActionController::TestCase
+  test "PATCH :update successfully updates the lead image and republishes the draft edition" do
+    login_as :writer
+
+    image = build(:image)
+    edition = create(:draft_case_study, images: [image])
+
+    Whitehall::PublishingApi
+    .expects(:save_draft)
+    .with(edition)
+    .returns(true)
+    .once
+
+    get :update, params: { edition_id: edition.id, id: image.id }
+
+    assert_equal image, edition.reload.lead_image
+    assert_redirected_to admin_edition_images_path(edition)
+    assert_equal "Lead image updated to minister-of-funk.960x640.jpg", flash[:notice]
+  end
+
+  test "PATCH :update does not update the lead image when the edition is invalid " do
```

Comment:
Is this a useful test case? There's no way to make an edition invalid via the lead images controller is there?

Reply:
I think so just because the change note validation error triggers a lot, so in that situation we want to make sure the action redirects correctly and populates the flash with the error.

Reply:
Ah so basically it's possible to arrive on this page with the edition in an invalid state because it is missing a change note?

Reply:
Yea unfortunately i think that's a reasonably common occurrence. The other option here could be to test the user tires to update the lead image to an image with the markdown in the body it handles that successfully. Both of those situations should trigger an unsuccessful save. I'll add another test for that.